### ISSUE TITLE
Nextcloud Slow until I disable collabra

### ISSUE THREAD

**Issue by leier69** [nextcloud/docker#1434]
Nextcloud is painfully slow until I disable collabra. Why is this? and please reffer me to the correct issues If i missed it before opening this one.

**Comment by leier69** [comment]
Yea I can also see that in the docker logs this shows up when trying to open collabra.
```
127.0.0.1 - - [03/Mar/2021:23:14:47 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
127.0.0.1 - - [03/Mar/2021:23:14:48 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
127.0.0.1 - - [03/Mar/2021:23:14:49 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
127.0.0.1 - - [03/Mar/2021:23:14:50 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
127.0.0.1 - - [03/Mar/2021:23:14:51 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
127.0.0.1 - - [03/Mar/2021:23:14:52 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
```

**Comment by leier69** [comment]
I also tried to run nextcloud in the host network. This seems to solve some people's issues, but not mine.

This is quite frustrating as nextcloud is ALMOST perfect. aww man

**Comment by WhiteBahamut** [comment]
Which version of Collabora you are running? Collabora Online - Built-in CODE Server? This destroyed performance for me as well. Running a CODE server in own container solved what ever slowed the connection down for me (though you need x86/x64 container run-time).

**Comment by leier69** [comment]
@WhiteBahamut How did you do that? Could you post ur compose?

**Comment by WhiteBahamut** [comment]
Only have k8s deployment file :) but I get most info from https://www.collaboraoffice.com/code/docker/ 
Once your CODE server is running, you need do configure it in the Collabora app in NC.

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: collabora-code
  namespace: YOURNAMESPACE
spec:
  selector:
    matchLabels:
      app: collabora-code
  template:
    metadata:
      labels:
        app: collabora-code
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/arch
                operator: In
                values:
                - "amd64"
      containers:
      - name: collabora-code
        image: collabora/code:6.4.6.2
        imagePullPolicy: IfNotPresent
        env:
        - name: domain
          value: "your.domain"  
        - name: username
          value: "admin" 
        - name: password
          value: "secert123"        
        ports:
        - containerPort: 9980
          protocol: TCP
        resources:
          requests:
            memory: "750Mi"
            cpu: "1000m"
          limits:
            memory: "2000Mi"
            cpu: "1750m"
        volumeMounts:
        - name: config
          mountPath: /etc/loolwsd
      volumes:
      - configMap:
          name: configs
          items:
            - key: loolwsd.xml
              path: loolwsd.xml              
        name: config
```

**Comment by leier69** [comment]
I get alot of these while its loading. Its clear that collabora-code fails, but it should be included in the nextcloud:21.0-apache image right?

```
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:46 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:47 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:48 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:49 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:50 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:51 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:52 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:53 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
app_1   | 127.0.0.1 - - [05/Mar/2021:14:00:54 +0000] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.38 (Debian) PHP/7.4.15 (internal dummy connection)"
```

**Comment by WhiteBahamut** [comment]
From my understanding it is NOT a part of the NC image. You have to install https://apps.nextcloud.com/apps/richdocuments. And then you need to setup the app, there you configure you collabora server location.
- Point to a real running server (what I did)
- use https://apps.nextcloud.com/apps/richdocumentscode or https://apps.nextcloud.com/apps/richdocumentscode_arm64 (which resulted in veeeery slow NC instance for me)
- use some demo server (never tried this)

**Comment by leier69** [comment]
Okai, I understad that its me that have been a retard on this one. BUT @WhiteBahamut how do you connect the two when you hos the collabora/code yourself?

**Comment by WhiteBahamut** [comment]
1. Follow instructions here https://www.collaboraoffice.com/code/docker/ to setup your own CODE server. I had to mount `/etc/loolwsd/loolwsd.xm` with my NC domain hardcoded in it.
2. Setup reverse proxy with your domain for collabora
3. Install https://apps.nextcloud.com/apps/richdocuments in NC
4. Configure richdocuments and select the first option ("Use your own server") and enter your collabora domain.
5. Click save and taadaaa it should work ;)

**Comment by bjo81** [comment]
> From my understanding it is NOT a part of the NC image. You have to install https://apps.nextcloud.com/apps/richdocuments. And then you need to setup the app, there you configure you collabora server location.

Usually the builtin CODE-Server is installed if you choose "install recommended apps". I got the issue that on a local instance all went fine and on a testbox the nginx/fpm-setup ran into 504 gateway timeouts on nearly every click.

**Comment by bjo81** [comment]
> I also tried to run nextcloud in the host network. This seems to solve some people's issues, but not mine.

Do you have an explanation why it should work better in the host network or a source for this assumption?

**Comment by WhiteBahamut** [comment]
> > From my understanding it is NOT a part of the NC image. You have to install https://apps.nextcloud.com/apps/richdocuments. And then you need to setup the app, there you configure you collabora server location.
> 
> Usually the builtin CODE-Server is installed if you choose "install recommended apps". I got the issue that on a local instance all went fine and on a testbox the nginx/fpm-setup ran into 504 gateway timeouts on nearly every click.

Ohh... never used the recommended apps. Which architecture are you using in your setups? I had the poor performance on ARM.

**Comment by leier69** [comment]
This was really wierd. I could not figure anything out.

So basically using docker swarm, and traefik the nextcloud image was really fucking slow. even when dropping traefik it did not make a difference. Uninstalling a lot of apps made the nextcloud instance a bit faster (mainly buildtin collabora).

Now I'm running my servers on kubernetes instead and the image is way faster. I wonder if the root cause was my docker swarm network setup. Anyways I'm closing this question now. Thanks for all the wonderful contributions.

**Comment by keponk** [comment]
@leier69 been looking for 2 weeks trying to figure out why the hell does it timeout, nothing on the main logs. Finally arrived to this thread and raised a new stack with my docker-compose and for the first time I tried not ticking the "install recommended apps". My instance is now as fast as I expected without any timeouts. THANKS! 
I was able to manually install the recommended apps and keep the speed (no timeouts), that is without including CODE. I suspect this might be the culprit. Not sure what happens behind the scenes while CODE is getting installed but I can imagine some internal DNS setup is raised as well causing some problems with docker.

Extra curiosity, this happened with an fpm/alpine setup. I also had an apache/debian setup where these default apps weren't an issue. I did noticed certain "lag" with the apache/debian which may or may not have been caused by the same default apps.
